Bottom: 7833129837a6aecead2559a649c1428d290a99f9
Top:    0963bedab22c1979762246b37507bc64422f6716
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-05-14 01:46:35 +0100

Refresh of major-re-write-to-drawing-rout

---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index f1a4085..290b00b 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -1689,6 +1689,7 @@ via_hole_cyl_callback (const BoxType * b, void *cl)
 void
 ghid_draw_everything (BoxTypePtr drawn_area)
 {
+  render_priv *priv = gport->render_priv;
   int i, ngroups;
   int side;
   /* This is the list of layer groups we will draw.  */
@@ -1703,7 +1704,7 @@ ghid_draw_everything (BoxTypePtr drawn_area)
   int min_phys_group;
   int max_phys_group;
 
-  current_color = NULL;
+  priv->current_colorname = NULL;
 
   /* Test direction of rendering */
   /* Look at sign of eye coordinate system z-coord when projecting a
